File: Test Case Generation/deepseek/cpp/test_task_15.cpp
#include <gtest/gtest.h>
#include <string>

using namespace std;

// Declaration of the function to test
string string_sequence(int n);

// Test suite for string_sequence function
class StringSequenceTest : public ::testing::Test {
protected:
    void SetUp() override {}
    void TearDown() override {}
};

// Test case: n = 0 (minimum value)
TEST_F(StringSequenceTest, HandlesMinimumValueZero) {
    EXPECT_EQ(string_sequence(0), "0");
}

// Test case: n = 1
TEST_F(StringSequenceTest, HandlesOne) {
    EXPECT_EQ(string_sequence(1), "0 1");
}

// Test case: n = 5 (normal case from example)
TEST_F(StringSequenceTest, HandlesFive) {
    EXPECT_EQ(string_sequence(5), "0 1 2 3 4 5");
}

// Test case: n = 10 (medium range)
TEST_F(StringSequenceTest, HandlesTen) {
    EXPECT_EQ(string_sequence(10), "0 1 2 3 4 5 6 7 8 9 10");
}

// Test case: n = 100 (larger number)
TEST_F(StringSequenceTest, HandlesHundred) {
    string expected = "0";
    for (int i = 1; i <= 100; ++i) {
        expected += " " + to_string(i);
    }
    EXPECT_EQ(string_sequence(100), expected);
}

// Test case: n = 999 (edge case - three digits)
TEST_F(StringSequenceTest, HandlesThreeDigitNumber) {
    string expected = "0";
    for (int i = 1; i <= 999; ++i) {
        expected += " " + to_string(i);
    }
    EXPECT_EQ(string_sequence(999), expected);
}

// Test case: n = -1 (negative number - the loop won't execute)
TEST_F(StringSequenceTest, HandlesNegativeValue) {
    EXPECT_EQ(string_sequence(-1), "0");
}

// Test case: n = -100 (large negative value)
TEST_F(StringSequenceTest, HandlesLargeNegativeValue) {
    EXPECT_EQ(string_sequence(-100), "0");
}

// Test case: n = INT_MAX (maximum integer value - verifies no overflow)
TEST_F(StringSequenceTest, HandlesMaxInt) {
    string expected = "0";
    // Only test up to 1000 for performance, verify the loop condition works
    EXPECT_NO_THROW(string_sequence(1000));
}

// Test case: verifying correct spacing pattern
TEST_F(StringSequenceTest, VerifiesCorrectSpacingPattern) {
    string result = string_sequence(3);
    // Check that there are no double spaces or leading/trailing spaces
    EXPECT_EQ(result, "0 1 2 3");
    EXPECT_EQ(result.find("  "), string::npos); // No double spaces
    EXPECT_NE(result.front(), ' '); // No leading space
    EXPECT_NE(result.back(), ' '); // No trailing space
}

// Test case: n = 2 (smallest non-trivial case)
TEST_F(StringSequenceTest, HandlesTwo) {
    EXPECT_EQ(string_sequence(2), "0 1 2");
}

// Edge case: n = INT_MIN verify no issues
TEST_F(StringSequenceTest, HandlesMinInt) {
    EXPECT_EQ(string_sequence(INT_MIN), "0");
}

File: Test Case Generation/deepseek/python/test_task_15.py
import pytest
from your_module import string_sequence

def test_zero():
    """Test with n = 0"""
    assert string_sequence(0) == '0'

def test_positive_number():
    """Test with a typical positive integer"""
    assert string_sequence(5) == '0 1 2 3 4 5'

def test_one():
    """Test with n = 1"""
    assert string_sequence(1) == '0 1'

def test_two():
    """Test with n = 2"""
    assert string_sequence(2) == '0 1 2'

def test_large_number():
    """Test with a larger number to ensure correct spacing"""
    result = string_sequence(10)
    expected = '0 1 2 3 4 5 6 7 8 9 10'
    assert result == expected

def test_negative_number():
    """Test with negative integer - should still work as range handles negatives"""
    result = string_sequence(-5)
    expected = ' '.join([str(x) for x in range(-5 + 1)])
    assert result == expected

def test_very_large_number():
    """Test with a very large number to ensure performance and correctness"""
    n = 1000
    result = string_sequence(n)
    expected = ' '.join([str(x) for x in range(n + 1)])
    assert result == expected

def test_return_type():
    """Test that the function always returns a string"""
    assert isinstance(string_sequence(0), str)
    assert isinstance(string_sequence(1), str)
    assert isinstance(string_sequence(100), str)
    assert isinstance(string_sequence(-10), str)

def test_no_trailing_spaces():
    """Test that there are no trailing spaces in the output"""
    result = string_sequence(5)
    assert not result.endswith(' ')
    assert result[-1] != ' ' if result else True

def test_no_leading_spaces():
    """Test that there are no leading spaces in the output"""
    result = string_sequence(5)
    assert not result.startswith(' ')
    assert result[0] != ' ' if result else True

def test_consecutive_numbers():
    """Test that numbers are consecutive and properly spaced"""
    result = string_sequence(10)
    numbers = [int(x) for x in result.split()]
    assert numbers == list(range(11))
    assert all(numbers[i] + 1 == numbers[i+1] for i in range(len(numbers)-1))

def test_string_format():
    """Test the exact string format with multiple numbers"""
    n = 3
    result = string_sequence(n)
    expected = '0 1 2 3'
    assert result == expected
    assert result.count(' ') == n  # n+1 numbers produce n spaces

def test_boundary_large_range():
    """Test boundary with maximum reasonable number"""
    n = 10000
    result = string_sequence(n)
    assert isinstance(result, str)
    assert len(result.split()) == n + 1